### PR TITLE
Refresh

### PR DESCRIPTION
**Pull request by dmt0**
- Updates to dev tools - SASS, Percy/Storybook, others
- Fix CI
- Drop IE support
- Use Volta
- Allow Plotly.js V2 as peer dep
- Fix minor bugs
- Minify production build
- Update demo app (used in github pages)
- Closes https://github.com/plotly/react-chart-editor/issues/1168
- Closes https://github.com/plotly/react-chart-editor/issues/1009
- Closes https://github.com/plotly/react-chart-editor/issues/854

### CODE REVIEW

**Comment by dmt0** [comment]
Hey @alexcjohnson, this one is ready for review. Once this is reviewed and merged I'll make a release.

Looks like Percy didn't keep the old screenshots and counts these as brand new initial ones. Would be great if you'd approve them, so we have a reference for future work.

File: .circleci/config.yml
Comment (alexcjohnson) [inline .circleci/config.yml:6]: Thanks for updating to `cimg`. But node 14 is still pretty old, most of our projects have moved on to at least 16, some to 18 (and now I see 20 is out even).

File: babel.config.json
Comment (alexcjohnson) [inline babel.config.json:31]:
```suggestion
```
or uncomment, whichever you actually intended.

File: src/styles/_movement.scss
Comment (alexcjohnson) [inline src/styles/_movement.scss:45]: where is `%animate` used? (and thanks for teaching me about [placeholder selectors](https://sass-lang.com/documentation/style-rules/placeholder-selectors/), new concept to me! 🧑‍🏫 )

File: package.json
Comment (alexcjohnson) [inline package.json:143]:
```suggestion
    "last 8 years and not dead"
```
I'm not quite sure what `"defaults"` means, but as I recall it was a little more aggressive about dropping older browsers than we were comfortable with in some contexts. The above is what we switched to in eg [dash-renderer](https://github.com/plotly/dash/blob/6eaf2e17c25f7ca1847c41aafeb18e87c586cb9f/dash/dash-renderer/package.json#L90) when we dropped IE support.

**Comment by alexcjohnson** [comment]
@dmt0 this looks great. Just had a couple of small suggestions, and it looks like whatever the issue was with Percy has been resolved and it's happy now ☺️

File: src/styles/_movement.scss
Comment (dmt0) [reply src/styles/_movement.scss:45]: You have the original developer of the feature to thank :)
We currently only use it to fade out modals on close, but there have been plans to animate more parts of the UI.

File: .circleci/config.yml
Comment (dmt0) [reply .circleci/config.yml:6]: I totally agree that 14 is old. IIRC this was an easy upgrade and upgrading to anything newer turned out to be non-trivial. I figured it doesn't matter that much since it's only used for CI. I can sure take a look at it though.

Comment (dmt0) [reply .circleci/config.yml]: Old Storybook doesn't want to work with newer Node. I've started the upgrade process. It's indeed non-trivial.

Comment (alexcjohnson) [reply .circleci/config.yml]: ah ok - would be nice to update for sure but if this is too onerous I won't insist.

File: src/styles/_movement.scss
Comment (alexcjohnson) [reply src/styles/_movement.scss:45]: I was just expecting to see an `@extend %animate` along with this, but I don't, am I misunderstanding how `%animate` would be used?

File: .circleci/config.yml
Comment (dmt0) [reply .circleci/config.yml]: Yeah, I'd argue it's best to delay this upgrade until https://github.com/storybookjs/storybook/issues/9828 is resolved. That's supposed to happen with Storybook V8 release.

As of now I've sunk a couple of hours into getting Storybook V7 to work. The result is pretty ugly (you can see it in my last commit). And now that Storybook works, Percy refuses to work, so I opened a support ticket with Percy to sort it out.

Node version in CI doesn't affect us much, as I haven't yet bumped into any packages that dropped support for it (except for Storybook).

I can take that Storybook commit and cherry-pick it into a separate PR in order not to loose that work.

File: src/styles/_movement.scss
Comment (dmt0) [reply src/styles/_movement.scss:45]: You're right! Animation is actually not there. I was using automated tools to do most of the SASS upgrade and haven't noticed how the animations disappeared. Fix incoming.

File: .circleci/config.yml
Comment (dmt0) [reply .circleci/config.yml]: Percy fixed by downgrading it. So we have a build. I'll let you decide if you wanna keep this Node + Storybook upgrade.

Comment (alexcjohnson) [reply .circleci/config.yml]: Thanks for fighting through that - let's keep to the latest Node + Storybook, seems like that's the bigger change than having to push Percy back a minor and will keep us closer to the latest versions overall.

**Review by alexcjohnson — APPROVED** [review]
💃 great stuff @dmt0!

**Pull request merged**
